assets/i18n/en.json

```diff
@@ -245,8 +245,8 @@
     "I18N_CONTACT_PAGE_GREETING_HEADING": "We would love to hear from you!",
     "I18N_CONTACT_PAGE_HEADING": "Contact Us",
     "I18N_CONTACT_PAGE_OTHER_INQUIRIES_HEADING": "Other Inquiries",
-    "I18N_CONTACT_PAGE_OTHER_INQUIRIES_PARAGRAPH_1": "For general questions, please reach out to <a href=\"mailto:admin@oppia.org\">admin@oppia.org</a>.",
-    "I18N_CONTACT_PAGE_OTHER_INQUIRIES_PARAGRAPH_2": "If you have a media inquiry, please send it to <a href=\"mailto:press@oppia.org\">press@oppia.org</a>.",
+    "I18N_CONTACT_PAGE_OTHER_INQUIRIES_PARAGRAPH_1": "For general questions, please reach out to <a class=\"oppia-contact-mail\" href=\"mailto:admin@oppia.org\">admin@oppia.org</a>.",
+    "I18N_CONTACT_PAGE_OTHER_INQUIRIES_PARAGRAPH_2": "If you have a media inquiry, please send it to <a class=\"oppia-contact-mail\" href=\"mailto:press@oppia.org\">press@oppia.org</a>.",
```

Comment:
@seanlip @DubeySandeep the reason for the class not applying for this may be that the content is there in the en.json file

Reply:
@Patel-Muhammad I don't understand what you mean. Could you please elaborate on the reasoning?

Reply:
@seanlip if you see line 248: "For general questions, please reach out to <a class=\"oppia-contact-mail\" href=\"mailto:admin@oppia.org\">admin@oppia.org</a>", here the class "oppia-contact-mail"  is given. And this content is there in the en.json file, so it is shown to the user by using : I18N_CONTACT_PAGE_OTHER_INQUIRIES_PARAGRAPH_1, I think that may be the reason that the class(oppia-contact-mail) is not applying if I put it in contact page related css.

Reply:
Hi @DubeySandeep please take a look at the comment and let me know if it is still an issue.

Reply:
Hi @DubeySandeep @seanlip can you please take a look at the reply and let me know if you have any concerns?

Reply:
Would you mind sharing the following devtools screenshot with the new changes:
![image](https://github.com/oppia/oppia/assets/16653571/4c30d251-bd26-46c4-bce9-a45bdb71b8f2)

This will help us understand which style is getting applied.

Reply:
![PHOTO-2023-10-24-21-59-23](https://github.com/oppia/oppia/assets/68547101/d55a8fb3-8828-42a4-bb52-fac598cf518c)

`text-decoration: none` is being applied currently. @Patel-Muhammad Please add devtools screenshot with the new changes. Added this for more clearity

Reply:
@DubeySandeep I have tried multiple approaches and this is the summary for each:

I have tried three approaches to solve this error (all failed):
1.
```
  .oppia-contact-page-content p a {
    text-decoration: underline;
  }
```
In the file: `oppia/core/templates/pages/contact-page/contact-page.component.html`

2.Adding inline style to the a tag:
```
 "I18N_CONTACT_PAGE_OTHER_INQUIRIES_PARAGRAPH_1": "For general questions, please reach out to <a href=\"mailto:[admin@oppia](mailto:admin@oppia).org\" style=\"text-decoration: underline;\">[admin@oppia.org](mailto:admin@oppia.org)</a>."

```

I have given inline style underline but this is also not working(the style is not applied, as seen from the inspect tab)

3. Adding a class to the a tag and targeting it in the contact page (oppia/core/templates/pages/contact-page/contact-page.component.html)
 ```
   "I18N_CONTACT_PAGE_OTHER_INQUIRIES_PARAGRAPH_1": "For general questions, please reach out to <a class=\"oppia-contact-mail\" href=\"mailto:[admin@oppia.org](mailto:admin@oppia.org)\">[admin@oppia.org](mailto:admin@oppia.org)</a>."

```
And I targeted this class in the contact-page.component.html:
```
.oppia-contact-mail{
    text-decoration: underline;
  }
```
This is also not working. As seen from the inspect page, the class is applied to the a tag but the style does not apply.
this is the devtools ss for all above 3 cases:
![Screenshot from 2023-10-27 10-23-58](https://github.com/oppia/oppia/assets/96219910/ca55f7b6-0b90-4445-816e-88af45493ca4)


In addition the tag has a style form third_party.css to make text-underline as none, that I can override if the class is targeted in oppia.css.
For conclusion the only way it is working is targeting the class in the oppia.css
See this screenshot the style is getting applied on if i put it in oppia.css
![Screenshot from 2023-10-27 10-25-41](https://github.com/oppia/oppia/assets/96219910/8c7850bd-3549-4a09-8b57-43675bd172a8)

Reply:
@Patel-Muhammad Have you tried a more specific CSS selector, e.g. `a.oppia-contact-mail`? In general specific selectors "win" over more general ones, and it seems like you have two selectors that are competing for the same element here. (I don't think it has anything to do with the class being specified in en.json like you mentioned previously.)

Reply:
Hi @seanlip I think there is some misunderstanding, please see the point 3. in the thread above, I have used the specific selector but it does not work.

Reply:
@Patel-Muhammad @Rijuta-s Thanks for sharing the screenshot!

@seanlip Thanks for the suggestion, I also think this will help us fix the issue.

@Patel-Muhammad Can you please follow [Sean's comment](https://github.com/oppia/oppia/pull/18815#issuecomment-1784109850) and fix it? I'll be happy to approve and merge the PR once it's resolved!

Reply:
@seanlip @DubeySandeep  initially I mis-read your suggestion, I tried adding a.oppai-contact-mail in contact page component, but the style still does not apply:
![Screenshot from 2023-10-31 12-25-28](https://github.com/oppia/oppia/assets/96219910/a68a1645-2cb1-4753-af95-0a911671c242)
![Screenshot from 2023-10-31 12-25-54](https://github.com/oppia/oppia/assets/96219910/525672ff-2b5e-4d9c-9712-728034d7c186)

But if i do the same by defining the style in oppia.css it is working:
![Screenshot from 2023-10-31 12-28-32](https://github.com/oppia/oppia/assets/96219910/902c5647-7439-46cd-a644-10a242b708b1)
![Screenshot from 2023-10-31 12-28-20](https://github.com/oppia/oppia/assets/96219910/57dbb0ac-a95f-4b49-8a0a-6523a9c6583b)


Let me know what should be done. Thanks!

Reply:
There are CSS specificity priority rules that determine which CSS rule takes precedence. Try and read them, and use those to figure out what to do next -- we can't give you the answer directly since this needs a bit of digging.

You can also try ruling out certain things. E.g. if you think that putting classes in the contact-page component or using en.json causes issues in itself, try applying a `color: red` class (say) and see what happens. If it doesn't apply at all then see if it applies for other CSS selectors in the contact page component. That way we can rule out or confirm the hypothesis that putting CSS in the contact page component doesn't work at all.

If the `color: red` CSS shows up then this suggests it's an issue where two selectors are competing for the same element and you'll need to use the CSS specificity rules to figure out what to do. Maybe making the selector even more specific will help.

Reply:
@seanlip I did through debugging as this is the approach and result:

I tried to add a sample paragraph tag and added a anchor tag with class oppia-contact-mail in the contact page component: (I did this because this is similar to the paragraph of the mail that is there in the page)
![Screenshot from 2023-11-03 13-10-59](https://github.com/oppia/oppia/assets/96219910/ac32120e-d03a-41c1-a331-726f78cae088)
The style  `.oppia-contact-mail` is also defined in the contact page component.

**What should have happened?**
The color of all anchor tags with the style `oppia-contact-mail` should be red and underlined.

**What actually happened?**
Only the style of the sample paragraph tag that I had added for testing, was seen red and underlined:
What should have happened?
![Screenshot from 2023-11-03 13-10-22](https://github.com/oppia/oppia/assets/96219910/9f37d775-1ba1-43d8-b7e7-bf0d66486c2b)

What is the inference?
This confirms that when class oppia-contact-mail is defined in the contact page then the anchor tag with class in defined in en.json then the class stylings does not apply to it.

@seanlip does this help?

Reply:
Great, that's excellent and much more helpful. Searching for this error brings this up https://stackoverflow.com/a/44215795 and the last two comments in the thread below the solution are probably worth testing (don't use ViewEncapsulation: None).

I know you've tested using oppia.css already and we can fallback to that (now that we have a proper rationale for it). But see if the ng-deep one works since that's more targeted (if it does work, add a comment pointing back to the SO thread and explaining what is going on so that future devs understand the context).

Thanks!

Reply:
Hi @seanlip, thanks for searching and showing the way forward! By using `::ng-deep .oppia-contact-mail` the style is being applied to the anchor tag now:
![Screenshot from 2023-11-03 17-01-25](https://github.com/oppia/oppia/assets/96219910/c6bdcb3f-0309-48b2-b4d0-9174fde957d1)

I have pushed the changes PTAL. Thanks!